### PR TITLE
Update api auth docs

### PR DESCRIPTION
**Pull request by vkrmis**
**Story card:** https://www.pivotaltracker.com/story/show/172643580
## Because

Our existing API authentication documentation is very sparse and doesn't describe the authentication flow.

## This addresses

Adds documentation for both User and Patient API authentication.

### CODE REVIEW

File: config/locales/en.yml [hunk 1]
```diff
@@ -84,7 +84,22 @@ en:
...
+
+        On successful verification of these credentials, the client will receive an access token. All subsequent requests to protected API endpoints should include this access token.
+
+        A successful authentication flow involves the following steps:
+        1. Submit the users phone number to the [`/v4/users/find` endpoint](#tag/User/paths/~1users~1find/post). If a user exists with this phone number, the response will return their `id`.
+        2. Submit the user's `id` retreived in the previous step, along with the user's `password` to the [`v4/users/activate` endpoint](#tag/User/paths/~1users~1activate/post). This will validate the user's password, and trigger an OTP to be sent to the user's phone number
+        3. Submit the user's `id`, their `password`, and the `otp` received after the previous step to the [`v3/login` endpoint](#tag/User-Login/paths/~1login/post). If the submitted otp and password are valid, this will return an `access_token` that can be used to authenticate the user
+
+        This token is a `Bearer` token, and should be attached as a header to all subsequent API requests as described here [link to bearer token docs here].
```
Comment (harimohanraj89) [inline config/locales/en.yml:100]: @vkrmis do we need to add info about the patient and facility ID headers too?

File: config/locales/en.yml [hunk 2]
```diff
@@ -81,14 +81,64 @@ en:
...
 
+        ## API Authentication
+        There are two types of authentication available to access different parts of the Simple API.
+        - User Authentication - For medical professionals using the Simple App. Grants access to most of the Simple API
+          to read and write data for communities of patients.
+        - Patient authentication - For individual patients. Grants access to read a patient's own data.
+
+        ### User Authentication
+
+        A Simple client can make authenticated requests to the Simple API on behalf of a medical professional with the
```
Comment (rsanheim) [inline config/locales/en.yml:99]: It sounds like all these creds are required .. can we be explicit about that? Maybe something like:

```
A client can make requests to the Simple API on behalf of a medical professional with the following three required credentials:
```

File: config/locales/en.yml [hunk 3]
```diff
@@ -81,14 +81,64 @@ en:
...
+
+        In order to obtain these credentials, a Simple client can authenticate a user with the following steps. The
+        client will need the user's registered phone number to perform the authentication.
+        1. Submit the users phone number to the [`POST /v4/users/find` endpoint](#tag/User/paths/~1users~1find/post). If
+        a user exists with this phone number, the response will return their id.
+        2. Submit the user's id retreived in the previous step, along with the user's password to the
+        [`POST v4/users/activate` endpoint](#tag/User/paths/~1users~1activate/post).
+        This will validate the user's password, and trigger an OTP to be sent to the user's phone number.
+        3. Submit the user's id, their password, and the OTP (received after the previous step) to the
+        [`POST v3/login` endpoint](#tag/User-Login/paths/~1login/post). If the submitted otp and password are valid,
```
Comment (rsanheim) [inline config/locales/en.yml:113]: How long is the access token valid for?  Does a client need to login on a some periodic basis to refresh the token?

File: config/locales/en.yml [hunk 4]
```diff
@@ -81,14 +81,64 @@ en:
...
+
+        ### Patient Authentication
+        A Simple client can make authenticated requests to the Simple API on behalf of a patient with the following
+        credentials:
+        - API Access Token
+        - Patient ID
+
+        In order to obtain these credentials, a Simple client can authenticate a patient with the following steps. The
+        client will need the patient's BP Passport UUID to perform the authentication.
+        1. Submit the patient's BP Passport UUID to the [`POST /v4/patients/activate` endpoint](#tag/Patient/paths/~1patient~1activate/post)
```
Comment (rsanheim) [inline config/locales/en.yml:132]: Does this POST will trigger the OTP being sent? If so, we should state that here explicitly.

File: config/locales/en.yml [hunk 5]
```diff
@@ -81,14 +81,64 @@ en:
...
+        2. Submit the patient's BP Passport UUID and OTP (received after the previous step) to the [`POST /v4/patients/login` endpoint](#tag/Patient/paths/~1patient~1login/post)
+
+        On successful authentication, the client will receive a payload containing an API access token and a patient ID.
```
Comment (rsanheim) [inline config/locales/en.yml:135]: Does the access tokens expire after a certain amount of time?

File: config/locales/en.yml (hunk 4 above)
Comment (timcheadle) [reply config/locales/en.yml:132]: It does.

File: config/locales/en.yml (hunk 5 above)
Comment (timcheadle) [reply config/locales/en.yml:135]: Not yet, working on that. Will update this doc when it's implemented.

File: config/locales/en.yml (hunk 3 above)
Comment (harimohanraj89) [reply config/locales/en.yml:113]: It's valid until the user signs into another device. I'll add explicit info.

**Comment by harimohanraj89** [comment]
@timcheadle that sidebar link was pointing to an auto-generated heading that this PR now hides. I've hidden the sidebar item as well. Now, the API Auth section is only jumpable to from here:

Screenshot 2020-05-05 at 12 16 40 PM

@rsanheim @timcheadle Feedback addressed, ready for another round of 👀 and maybe ✅ 😅

**Pull request merged**